homeassistant/components/vicare/binary_sensor.py

```diff
@@ -191,11 +171,12 @@ class ViCareBinarySensor(ViCareEntity, BinarySensorEntity):
     def __init__(
         self,
         device_config: PyViCareDeviceConfig,
-        api: PyViCareDevice | PyViCareHeatingDeviceComponent,
+        device: PyViCareDevice,
+        component: PyViCareHeatingDeviceComponent | None,
         description: ViCareBinarySensorEntityDescription,
```

Comment:
Maybe consider swapping these two and make None the default? That looks a bit cleaner than just passing `None` in the super constructor as that doesn't really show what is None